app/components/assets/assets-index/advanced-asset-columns.tsx

```diff
@@ -331,11 +332,48 @@ function TextColumn({
   );
 }
 
-function StatusColumn({ status }: { status: AssetStatus }) {
+function StatusColumn({
```

Comment:
This is overall good but in this way it works only on advanced index. We should make it work in any context where the AssetStatusBadge is used.
Considering that the bookings wont always be available, we should make it so that if the bookings are not passed the component makes an api request to fetch the bookings for that specific asset so if they are not passed it still works. THat way we dont have to change the loaders on every view we show assets to also query the bookings. Then the query can also be made to happen only if the asset is checked out already.